### PR TITLE
[Documentation] Sorting and filtration in api

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no|
| BC breaks?      | no|
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: docs/api/sorting_and_filtration.rst [hunk 1]
```diff
@@ -0,0 +1,710 @@
...
+.. code-block:: text
+
+    STATUS: 200 OK
+
+.. code-block:: json
+
+    {
+        "page": 1,
+        "limit": 10,
+        "pages": 1,
```
Comment (lchrusciel) [inline docs/api/sorting_and_filtration.rst]: Are these values really correct?

File: docs/api/sorting_and_filtration.rst [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
+.. code-block:: text
+
+    GET /api/v1/products/?criteria\[{nameOfCriterion}\]\[type\]={searchOption}&criteria\[{nameOfCriterion}\]\[value\]={searchingPhrase}'
+
++-----------------+----------------+-------------------------------------------------------------+
+| Parameter       | Parameter type | Description                                                 |
++=================+================+=============================================================+
+| Authorization   | header         | Token received during authentication                        |
++-----------------+----------------+-------------------------------------------------------------+
+| nameOfCriterion | query          | The name of criterion (for example "search", "couponBased") |
```
Comment (lchrusciel) [inline docs/api/sorting_and_filtration.rst]: criterion can be `couponBased` or `search`?

Comment (tuka217) [inline docs/api/sorting_and_filtration.rst]: can be

File: docs/api/sorting_and_filtration.rst [hunk 3]
```diff
@@ -0,0 +1,610 @@
...
+
+In Sylius API list of resources can be sorted and filtered by passed url query parameters. Here you can find examples how to
+do it with sample resources.
+
+.. note::
+
+    To find out by which fields the api resources can be sorted and how they can be filtered you should check the grid configuration of these
+    `here <https://github.com/Sylius/Sylius/tree/master/src/Sylius/Bundle/ApiBundle/Resources/config/grids>`_
+
+How to sort resources ?
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: Space before `?`

File: docs/api/sorting_and_filtration.rst [hunk 4]
```diff
@@ -0,0 +1,610 @@
...
+-----------------------
+
+Let's assume that you want to sort products by code in descending direction. In these case you should call
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: * descending **order**
* In **this** case

File: docs/api/sorting_and_filtration.rst [hunk 5]
```diff
@@ -0,0 +1,610 @@
...
+.. code-block:: text
+
+    GET /api/v1/products/?sorting\[{nameOfField}\]={direction}'
+
++---------------+----------------+----------------------------------------------------------------+
+| Parameter     | Parameter type | Description                                                    |
++===============+================+================================================================+
+| Authorization | header         | Token received during authentication                           |
++---------------+----------------+----------------------------------------------------------------+
+| nameOfField   | query          | *(required)* Name of field by with the resource will be sorted |
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: by **which** the the...

File: docs/api/sorting_and_filtration.rst [hunk 6]
```diff
@@ -0,0 +1,610 @@
...
++---------------+----------------+----------------------------------------------------------------+
+| direction     | query          | *(required)* Define an direction of ordering                   |
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: **a** direction

File: docs/api/sorting_and_filtration.rst [hunk 7]
```diff
@@ -0,0 +1,610 @@
...
+                        "self": {
+                            "href": "\/api\/v1\/products\/e77f129f-5921-3ad2-88bd-f27b59aad037"
+                        }
+                    }
+                }
+            ]
+        }
+    }
+
+How to filter resources ?
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: Space before `?`

File: docs/api/sorting_and_filtration.rst [hunk 8]
```diff
@@ -0,0 +1,610 @@
...
+-------------------------
+
+Let's assume that you want to find all products which contain in name the word ``sticker``. In these case you should call
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: * which contain the word ``sticker`` in the name
* In this case

File: docs/api/sorting_and_filtration.rst [hunk 9]
```diff
@@ -0,0 +1,610 @@
...
++-----------------+----------------+--------------------------------------------------------------------------+
+| searchOption    | query          | *(required)* Option of searching (for example "contains", "equal")       |
++-----------------+----------------+--------------------------------------------------------------------------+
+| limit           | query          | *(optional)* Number of items to display per page,                        |
+|                 |                | by default = 10                                                          |
++-----------------+----------------+--------------------------------------------------------------------------+
+
+.. note::
+
+    The *nameOfCriterion* is a key from grid configuration of a sample resource.
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: from **the** grid..

File: docs/api/sorting_and_filtration.rst [hunk 10]
```diff
@@ -0,0 +1,610 @@
...
+|                 |                | by default = 10                                                          |
++-----------------+----------------+--------------------------------------------------------------------------+
+
+.. note::
+
+    The *nameOfCriterion* is a key from grid configuration of a sample resource. 
+
+.. tip::
+
+    You can find list of all search options in :doc:`GridBundle docs</bundles/SyliusGridBundle/filters>`.
```
Comment (venyii) [inline docs/api/sorting_and_filtration.rst]: You can find **a** list of...

File: docs/api/sorting_and_filtration.rst [hunk 11]
```diff
@@ -0,0 +1,610 @@
...
+.. note::
+
+    To find out by which fields the api resources can be sorted and how they can be filtered you should check the grid configuration of these
+    `here <https://github.com/Sylius/Sylius/tree/master/src/Sylius/Bundle/ApiBundle/Resources/config/grids>`_
+
+How to sort resources?
+----------------------
+
+Let's assume that you want to sort products by code in descending order. In this case you should call
+the ``/api/v1/products/`` endpoint with the ``GET`` method and sorting query parameters.
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: and **provide** sorting query parameters.

File: docs/api/sorting_and_filtration.rst [hunk 12]
```diff
@@ -0,0 +1,610 @@
...
+| Parameter     | Parameter type | Description                                                     |
++===============+================+=================================================================+
+| Authorization | header         | Token received during authentication                            |
++---------------+----------------+-----------------------------------------------------------------+
+| nameOfField   | query          | *(required)* Name of field by which the resource will be sorted |
++---------------+----------------+-----------------------------------------------------------------+
+| direction     | query          | *(required)* Define a direction of ordering                     |
++---------------+----------------+-----------------------------------------------------------------+
+| limit         | query          | *(optional)* Number of items to display per page,               |
+|               |                | by default = 10                                                 |
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: Do we need to put it in two lines?

File: docs/api/sorting_and_filtration.rst [hunk 13]
```diff
@@ -0,0 +1,610 @@
...
+                }
+            ]
+        }
+    }
+
+How to filter resources?
+------------------------
+
+Let's assume that you want to find all products which contain the word ``sticker`` in the name. In this case you should call
+the ``/api/v1/products/`` endpoint with the ``GET`` method and filter query parameters.
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: and **provide** filter query parameters.

File: docs/api/sorting_and_filtration.rst [hunk 14]
```diff
@@ -0,0 +1,610 @@
...
+    GET /api/v1/products/?criteria\[{nameOfCriterion}\]\[type\]={searchOption}&criteria\[{nameOfCriterion}\]\[value\]={searchingPhrase}'
+
++-----------------+----------------+--------------------------------------------------------------------------+
+| Parameter       | Parameter type | Description                                                              |
++=================+================+==========================================================================+
+| Authorization   | header         | Token received during authentication                                     |
++-----------------+----------------+--------------------------------------------------------------------------+
+| nameOfCriterion | query          | *(required)* The name of criterion (for example "search", "couponBased") |
++-----------------+----------------+--------------------------------------------------------------------------+
+| searchingPhrase | query          | *(required)* The searching phrase                                        |
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: I know it's not the best place to suggest that, but why isn't it just `searchPhrase`?

File: docs/api/sorting_and_filtration.rst [hunk 15]
```diff
@@ -0,0 +1,610 @@
...
++-----------------+----------------+--------------------------------------------------------------------------+
+| searchOption    | query          | *(required)* Option of searching (for example "contains", "equal")       |
++-----------------+----------------+--------------------------------------------------------------------------+
+| limit           | query          | *(optional)* Number of items to display per page,                        |
+|                 |                | by default = 10                                                          |
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: also here, why 2 lines instead of 1?

File: docs/api/sorting_and_filtration.rst [hunk 16]
```diff
@@ -0,0 +1,610 @@
...
+
+.. note::
+
+    The *nameOfCriterion* is a key from the grid configuration of a sample resource.
+
+.. tip::
+
+    You can find a list of all search options in :doc:`GridBundle docs</bundles/SyliusGridBundle/filters>`.
+
+
```
Comment (TheMadeleine) [inline docs/api/sorting_and_filtration.rst]: redundant line

**Comment by pjedrzejewski** [comment]
Thank you Ania!

**Pull request merged**